server.js

```diff
@@ -1,12 +1,65 @@
+//importing express
 const express = require("express");
 
-const port = process.env.PORT || 8000;
+//importing mongoose
+const mongoose = require("mongoose")
+
+//importing cors to resolve any CORS errors in the browser
+const cors = require("cors")
+
+//setting up default port
+const port = process.env.PORT || 5000;
+
+//binding express to a variable
 const app = express();
 
-const Message = require('./Message.js')
+//importing message schema
+const MessageSchema = require('./Message.js')
+
+//creating the intial connection to the database
+mongoose.connect("mongodb://localhost:27017/chat")
+
+//init the database through the connectino construstor, stored in a variable
+const db = mongoose.connection
+
+//binds error message to the connection variable to print if an error occurs
+db.on('error', console.error.bind(console, 'connection error'))
 
+//creating the entry model utilizing the entry schema and entries collection 
+const Entry = mongoose.model("entries", MessageSchema)
 
+//middleware functions
+app.use(express.static("./build"));
+app.use(express.urlencoded({ extended: true }));
+app.use(cors());
 
 app.listen(port, () => {
-  console.log('listening on port: ' + port) 
+  console.log(`Listening on port: ${port}`) 
 })
+
+//creating API route for the front end to access ALL the entries from the database
+app.get("/allentries", async (req, res) => {
```

Comment:
Try writing a GET request to enable specific messages to return to specific rooms. 
Hint: look into implementing request.params.